### ISSUE TITLE
cargo-deb takes extremly long to setup/install

### ISSUE THREAD

**Issue by Oekel** [IamTheCarl/system-mqtt#13]
Sorry, I can't find a form to send the distributors of that repro a Mail or similar.

So maybee you can help me out right here
https://community.openhab.org/t/https-github-com-iamthecarl-system-mqtt/154907

Not sure if it is based on a bug, but maybee we could discuss.

Thank you!

**Comment by IamTheCarl** [comment]
Unfortunatly, I do not have access to a Pi1 to test on, so I can only speculate what your issue is and how to fix it.

If I recall correctly, that version only has 512Mb of RAM. I've never tested builds on something that small but it's probably not enough. As a quick and dirty test, I ran `/usr/bin/time -v cargo build -j1` on my machine and found that it used about 583Mb of memory. That's certainly enough to start spilling into the swap space, or crash if no swap space is available. Note that this was on an x86 system, so it could be better or worse than what a 32bit ARM system would get.

I recommend you try cross-compiling from your host machine to the Pi.
I typically use [cross](https://github.com/cross-rs/cross) when doing one off builds for embedded devices.
I tried the command `cross deb --target armv7-unknown-linux-gnueabi` but unfortunately that failed due to an error [related to the ring crate](https://github.com/briansmith/ring/issues/1616#issuecomment-1645344009). It's work hours for me right now so I'll have to look more into how to fix that later.
